docs/source/developers/guide/resources.rst

```diff
@@ -27,3 +27,51 @@
 ************************************
 Additional information and resources
 ************************************
+
+On this page we have listed resources that may be relevant or useful for
+contributors who want to learn more about different parts of Apache Arrow.
+
+.. Annotation Glossary
+.. -------------------
+.. ARROW-15130
+
+Additional information
+----------------------
+
+- GitHub Actions
+
+  GitHub Actions is a continuous integration and continuous delivery (CI/CD) platform.
+  Apache Arrow has a set of workflows that run via GitHub Actions to build and test
+  every pull request that is opened, merged, etc.
+
+  - `Apache Arrow Actions on GitHub <https://github.com/apache/arrow/actions>`_
+  - `Location of the workflows in Arrow: arrow/.github/workflows/ <https://github.com/apache/arrow/tree/master/.github/workflows>`_
+  - `GitHub Documentation on GitHub Actions <https://docs.github.com/en/actions>`_
+  .. ARROW-13841: [Doc] Document the different subcomponents that make up the CI and how they fit together:
+  .. https://github.com/apache/arrow/pull/11821
+
+- Nightly builds
+
+  It is possible to install the Arrow R package from the nightly builds which are daily development
+  builds of the R package and are not the official releases. See more on the
+  `Install R package article <https://ursalabs.org/arrow-r-nightly/articles/install.html#install-the-nightly-build>`_.
+
+- `Apache Arrow releases <https://arrow.apache.org/release/>`_
+
+Other resources
+---------------
+
+- :ref:`contributing`
+- `Arrow R Developer Guide <https://arrow.apache.org/docs/r/articles/developing.html>`_
+- `Writing Bindings article for R package <https://arrow.apache.org/docs/r/articles/developers/bindings.html>`_.
+
+- `Tidyverse: Make a reprex <https://www.tidyverse.org/help/#reprex>`_
+- `Craft Minimal Bug Reports by Matthew Rocklin <https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports>`_
+
+Recommended books for learning languages
+----------------------------------------
+
+- Slatkin, Brett, *Effective Python: 90 Specific Ways to Write Better Python*, Addison-Wesley Professional, 2019
+- Stroustrup, Bjarne, *A Tour of C++ (Second edition)*, Addison-Wesley, 2018
+- Wickham, Hadley, `R Packages: Organize, Test, Document, and Share Your Code <https://r-pkgs.org/>`_
+- Wickham, Hadley, `Advanced R <https://adv-r.hadley.nz/>`_
```

Comment:
I wonder if this might be out of the scope of this guide?

Reply:
Perhaps, yes. There were, though, quite some suggestions to include the resources for earning languages from previous reviews of the guide. Not intended to learn a language from scratch, but to have a reference of which books are recommended by the Arrow developers (to be used as help with structuring the code). Does that makes sense?

Reply:
Yeah, that does make sense.  Perhaps could we just rename this section header then to reflect this?  Maybe "Recommended references" or something?

Reply:
Sure, will do it as soon as I get back to the computer.